Test failure in 0.40.0 on 64-bit PowerPC: test_get_process_triple

<!--

Thanks for opening an issue! To help the Numba team handle your information
efficiently, please first ensure that there is no other issue present that
already describes the issue you have
(search at https://github.com/numba/llvmlite/issues?&q=is%3Aissue).

-->

## Reporting a bug

<!--

Before submitting a bug report please ensure that you can check off these boxes:

-->

- [x] I have tried using the latest released version of llvmlite (most recent is
 visible in the change log (https://github.com/numba/llvmlite/blob/main/CHANGE_LOG).
- [ ] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as 'python bug.py'. **N/A**

<!--

Please include details of the bug here, including, if applicable, what you
expected to happen!

-->

I [just tested](https://src.fedoraproject.org/rpms/python-llvmlite/pull-request/5) updating the [`python-llvmlite` package](https://src.fedoraproject.org/rpms/python-llvmlite) in Fedora Linux to 0.40.0 (in Rawhide, the development version of Fedora).

I got exactly one test failure—only on `ppc64le`—as follows:

```
======================================================================
FAIL: test_get_process_triple (llvmlite.tests.test_binding.TestMisc.test_get_process_triple)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/llvmlite-0.40.0/llvmlite/tests/test_binding.py", line 587, in test_get_process_triple
    self.assertEqual(default_parts[0], triple_parts[0])
AssertionError: 'ppc64le' != 'powerpc64le'
- ppc64le
+ powerpc64le
?  ++++
```

What do you think? Thanks!

> I don't know how to test this, but I added normalization to the default triple: #948

When that PR is applied as a patch to the Fedora Linux package, the [tests still fail in the same way on `ppc64le`](https://koji.fedoraproject.org/koji/taskinfo?taskID=101249014):

```
======================================================================
FAIL: test_get_process_triple (llvmlite.tests.test_binding.TestMisc.test_get_process_triple)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/llvmlite-0.40.0/llvmlite/tests/test_binding.py", line 587, in test_get_process_triple
    self.assertEqual(default_parts[0], triple_parts[0])
AssertionError: 'ppc64le' != 'powerpc64le'
- ppc64le
+ powerpc64le
?  ++++
----------------------------------------------------------------------
```

--- a/llvmlite/tests/test_binding.py
+++ b/llvmlite/tests/test_binding.py
@@ -631,13 +631,20 @@ class TestMisc(BaseTest):
         self.assertTrue(triple)
 
     def test_get_process_triple(self):
+        # Sometimes we get synonyms for PPC
+        def normalize_ppc(arch):
+            if arch == 'powerpc64le':
+                return 'ppc64le'
+            else:
+                return arch
+
         triple = llvm.get_process_triple()
         default = llvm.get_default_triple()
         self.assertIsInstance(triple, str)
         self.assertTrue(triple)
 
-        default_parts = default.split('-')
-        triple_parts = triple.split('-')
+        default_parts = normalize_ppc(default.split('-'))
+        triple_parts = normalize_ppc(triple.split('-'))
         # Arch must be equal
         self.assertEqual(default_parts[0], triple_parts[0])
 

--- a/llvmlite/tests/test_binding.py
+++ b/llvmlite/tests/test_binding.py
@@ -643,10 +643,10 @@ class TestMisc(BaseTest):
         self.assertIsInstance(triple, str)
         self.assertTrue(triple)
 
-        default_parts = normalize_ppc(default.split('-'))
-        triple_parts = normalize_ppc(triple.split('-'))
+        default_arch = normalize_ppc(default.split('-')[0])
+        triple_arch = normalize_ppc(triple.split('-')[0])
         # Arch must be equal
-        self.assertEqual(default_parts[0], triple_parts[0])
+        self.assertEqual(default_arch, triple_arch)
 
     def test_get_host_cpu_features(self):
         features = llvm.get_host_cpu_features()

